src/psyclone/gocean1p0.py

```diff
@@ -328,33 +273,19 @@ def gen_code(self, parent):
         # Generate the code body of this subroutine
         self.schedule.gen_code(invoke_sub)
 
-        # If we're generating an OpenCL routine then the arguments must
-        # have the target attribute as we pass pointers to them in to
-        # the OpenCL run-time.
-        target = bool(self.schedule.opencl)
-
         # Add the subroutine argument declarations for fields
         if self.unique_args_arrays:
             my_decl_arrays = TypeDeclGen(invoke_sub, datatype="r2d_field",
-                                         intent="inout", target=target,
+                                         intent="inout",
                                          entity_decls=self.unique_args_arrays)
             invoke_sub.add(my_decl_arrays)
 
         # Add the subroutine argument declarations for integer and real scalars
-        r_args = []
         i_args = []
         for argument in self.schedule.symbol_table.argument_datasymbols:
```

Comment:
Now that OpenCL doesn't use the gen_code, this is practically dead code and was not covered.

The reason it is not fully deleted is because the psy_data tests call this gen_code directly but it is confusing because the psy_data transformations and pyclone psy_data command options will use the lowering+backend path and not the gen_code.
However, this direct gen_code tests can not be deleted neither because they touch the psygen gen_code that is necessary for the psy_data transformations when applied to LFRic. LFRic don't test this by itself.

I now added a TODO that says this is about to be deleted but I leave this for another PR as it has to modify the mentioned tests and this one is already big enough.